### PR TITLE
Add workaround for UEFI mode

### PR DESCRIPTION
**Pull request by RoyCai7**
POO: https://progress.opensuse.org/issues/119239
UEFI,gnome workaround for bug: https://bugzilla.suse.com/show_bug.cgi?id=1203641

- Related ticket:  https://progress.opensuse.org/issues/119239
- Needles: N/A
- Verification run: [legacy and gnome](http://openqa.suse.de/t9801030), [UEFI and gnome](http://openqa.suse.de/t9801031), [legacy and textmode](http://openqa.suse.de/t9801876), [UEFI and textmode](http://openqa.suse.de/t9802074)

### CODE REVIEW

**Comment by Julie-CAO** [comment]
If you would change to use get_var('UEFI'), 'UEFI' can be a optional setting for a test suite.

**Review by alice-suse — APPROVED** [review]
LGTM

**Comment by RoyCai7** [comment]
> If you would change to use get_var('UEFI'), 'UEFI' can be a optional setting for a test suite.
Good point, will cover it when UEFI is optional as legacy.

File: lib/opensusebasetest.pm [hunk 1]
```diff
@@ -1049,13 +1049,22 @@ sub wait_boot_past_bootloader {
...
         # This should only happen on SLE15SP5 and on hyperv
-        record_soft_failure 'workaround bsc#1204221 - Failed to boot at SLES15SP5 with gnome in hyperv 2019' if check_var('DESKTOP', 'gnome');
-        record_soft_failure 'workaround bsc#1204230 - Failed to boot at SLES15SP5 with x server in hyperv 2019' if check_var('DESKTOP', 'textmode');
-        sleep 30;
-        send_key 'esc';
+        # This is for legacy issue.
+        if (check_var('UEFI', '0')) {
+            record_soft_failure 'workaround bsc#1204221 - Failed to boot at SLES15SP5 with gnome in hyperv 2019' if check_var('DESKTOP', 'gnome');
+            record_soft_failure 'workaround bsc#1204230 - Failed to boot at SLES15SP5 with x server in hyperv 2019' if check_var('DESKTOP', 'textmode');
+            sleep 30;
```
Comment (waynechen55) [inline lib/opensusebasetest.pm:1059]: Using "sleep" is not optimal. You should avoid such usage unless it is absolutely necessary or you do not have other measures to achieve what you want to do. Please refer to [code contribution](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md)

File: lib/opensusebasetest.pm [hunk 2]
```diff
@@ -1049,13 +1049,22 @@ sub wait_boot_past_bootloader {
...
+            send_key 'esc';
+        }
+        # This is for UEFI issue.
+        if (check_var('UEFI', '1') && check_var('DESKTOP', 'gnome')) {
+            record_soft_failure 'workaround bsc#1203641 - Failed to boot after installation on hyperv-2019 UEFI setup';
+            sleep 30;
```
Comment (waynechen55) [inline lib/opensusebasetest.pm:1065]: Using "sleep" is not optimal. You should avoid such usage unless it is absolutely necessary or you do not have other measures to achieve what you want to do. Please refer to [code contribution](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md)

Comment (RoyCai7) [reply lib/opensusebasetest.pm:1065]: There is no better alternative, just use it.

File: lib/opensusebasetest.pm [hunk 3]
```diff
@@ -1049,13 +1049,22 @@ sub wait_boot_past_bootloader {
...
 
-    # Workaround for bsc#1204221 and bsc#1204230
-    if (is_sle('=15-SP5') && check_var('VIRSH_VMM_FAMILY', 'hyperv')) {
-        # This should only happen on SLE15SP5 and on hyperv
-        record_soft_failure 'workaround bsc#1204221 - Failed to boot at SLES15SP5 with gnome in hyperv 2019' if check_var('DESKTOP', 'gnome');
-        record_soft_failure 'workaround bsc#1204230 - Failed to boot at SLES15SP5 with x server in hyperv 2019' if check_var('DESKTOP', 'textmode');
-        sleep 30;
-        send_key 'esc';
+    # Workaround for bsc#1204221, bsc#1204230 and bsc#1203641
+    if (is_sle('=15-SP5') && check_var('VIRSH_VMM_FAMILY', 'hyperv') && check_var('HYPERV_VERSION', '2019') && check_var('FLAVOR', 'Online')) {
```
Comment (waynechen55) [inline lib/opensusebasetest.pm:1053]: Why you need this ?
```
check_var('FLAVOR', 'Online')
```

Comment (RoyCai7) [reply lib/opensusebasetest.pm:1053]: It should not include workaround for jobs: [snapper_jeos_cli/36](https://openqa.suse.de/tests/9823700#step/snapper_jeos_cli/36)
so filter it.

**Pull request merged**